src/components/main-view/main-view.jsx

```diff
@@ -2,14 +2,26 @@ import { useState, useEffect } from "react";
 import PropTypes from 'prop-types';
 import { MovieCard } from "../movie-card/movie-card";
 import { MovieView } from "../movie-view/movie-view";
+import { LoginView } from "../login-view/login-view";
+import { SignupView } from "../signup-view/signup-view";
 
 export const MainView = () => {
+    const storedUser = JSON.parse(localStorage.getItem("user"));
+    const storedToken = localStorage.getItem("token");
+    const [user, setUser] = useState(storedUser ? storedUser : null);
+    const [token, setToken] = useState(storedToken ? storedToken : null);
     const [movies, setMovies] = useState([]);
-
     const [selectedMovie, setSelectedMovie] = useState(null);
 
+
     useEffect(() => {
-        fetch("https://movie-nest-app-630a7e8ce836.herokuapp.com/movies")
+        if (!token) {
+            return;
+        }
+
+        fetch("https://movie-nest-app-630a7e8ce836.herokuapp.com/movies", {
+            headers: { Authorization: `Bearer ${token}`}
+        })
           .then((response) => response.json())
           .then((data) => {
             console.log("movies from api:", data);
```

Comment:
console.logs are great for debugging but kindly remove them in the final submission.

Reply:
console.log removed